neqo-common/src/datagram.rs

```diff
@@ -54,6 +54,7 @@ impl Datagram {
         self.ttl
     }
 
+    #[cfg(feature = "udp")]
```

Comment:
Would this warning be generated if we added `impl From<Datagram> for Vec<u8>` instead?

Reply:
Currently `Datagram::into_data` is `pub(crate)` as it is only used within `neqo-common`, more specifically the `neqo_common::udp` module. Replacing `Datagram::into_data` with `impl From<Datagram> for Vec<u8>` would make the functionality `pub`. All `pub` crate items are assumed to be used. Thus the warning would be gone, yes.

I suggest not exposing it as `pub`, as that exposes the internal data representation of `Datagram`. This is not a strong opinion. Happy to go with whatever you prefer.

(Tangentially, we might want to reconsider the data representation of `Datagram` as part of https://github.com/mozilla/neqo/issues/1693.)